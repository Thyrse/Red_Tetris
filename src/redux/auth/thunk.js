import { setUserData, setDisconnectUser, setUserDataError } from "./actions";

// THUNK

export const login = (user) => async (dispatch) => {
  console.log("USER SENT TO THUNK ==>", user);
  //   try {
  //     return auth;
  //   } catch (err) {
  //     console.error("AUTHENTICATE_ERROR", err);
  //     dispatch(setUserDataError(err.toString()));
  //   }
};
