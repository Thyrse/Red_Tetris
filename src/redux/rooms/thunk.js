export const hydrateStore = async (dispatch) => {
  // console.log("HYDRATESTORE");
  try {
    // fill store
    await Promise.all([
      dispatch(fetchWorkspaces()),
      dispatch(fetchChatrooms()),
    ]);
  } catch (e) {
    console.error("HYDRATESTORE_ERROR", e);
  }
};
