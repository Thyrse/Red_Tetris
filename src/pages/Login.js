/* eslint-disable array-callback-return */
// REACT
import React, { useState, useEffect, useContext, useCallback } from "react";
// REACT ROUTER
import { Link, useHistory } from "react-router-dom";
import socketIOClient from "socket.io-client";
import { useDispatch, useSelector } from "react-redux";
import { setUserData } from "../redux/auth/actions";

/**
 * Component that displays the patient page,
 * contains the list of patients,
 * as well as the components for display and editing,
 * drawers and modal
 */
const Login = () => {
  const socket = socketIOClient.connect("http://localhost:4000");
  const [username, setUsername] = useState();
  const history = useHistory();
  const dispatch = useDispatch();

  const handleChange = (e) => {
    setUsername(e.target.value);
  };
  console.log("SOCKET ==>", socket);
  const handleSubmit = (e) => {
    e.preventDefault();
    socket.emit("login", username.toUpperCase());
    dispatch(
      setUserData({ username: username.toUpperCase(), socketID: socket.id })
    );
    // dispatch(setUserData(username.toUpperCase()));
    history.push("/home");
  };

  console.log("USERNAME ==>", username);
  return (
    <>
      <div className="d-flex justify-content-center">
        <div className="login p-3 shadow rounded text-center text-white">
          <div>
            <h3>Connect to play!</h3>
          </div>
          <form onSubmit={handleSubmit}>
            <div className="m-3">
              <label>Username:</label>
              <input
                type="text"
                name="username"
                id="username"
                className="form-control"
                placeholder="Tefourge, Scao..."
                onChange={(e) => handleChange(e)}
              />
            </div>
            <div className="d-flex justify-content-center">
              <button type="submit" className="btn btn-white">
                Play
              </button>
            </div>
          </form>
        </div>
      </div>
    </>
  );
};

export default Login;
