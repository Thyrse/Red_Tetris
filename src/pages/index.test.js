import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Login from "./Login";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];

const mockStore = configureStore(middlewares);
const initialState = {
  userDatas: null,
  roomsList: [],
  usersList: null,
};

const store = mockStore(initialState);

describe("<Login />", () => {
  test("Login redering upon arrival", () => {
    const wrapper = shallow(<Login store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
