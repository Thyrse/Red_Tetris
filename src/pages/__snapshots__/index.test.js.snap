// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Login redering upon arrival 1`] = `
<Fragment>
  <div
    className="d-flex justify-content-center"
  >
    <div
      className="login p-3 shadow rounded text-center text-white"
    >
      <div>
        <h3>
          Connect to play!
        </h3>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="m-3"
        >
          <label>
            Username:
          </label>
          <input
            className="form-control"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Thyrse, Ziphlot..."
            type="text"
          />
        </div>
        <div
          className="d-flex justify-content-center"
        >
          <button
            className="btn btn-white"
            type="submit"
          >
            Play
          </button>
        </div>
      </form>
    </div>
  </div>
</Fragment>
`;
